src/App.tsx

```diff
@@ -1,14 +1,70 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterTypes } from './types/FilterTypes';
+
+function filterTodos(todos: Todo[], todosActivityFilter: FilterTypes, todosQuery: string) {
+  let resultTodos = [...todos];
+  const query = todosQuery.toLowerCase();
+
+  switch (todosActivityFilter) {
+    case 'completed':
+      resultTodos = resultTodos.filter(todo => todo.completed
+      && todo.title.toLowerCase().includes(query));
+      break;
+    case 'active':
+      resultTodos = resultTodos.filter(todo => !todo.completed
+      && todo.title.toLowerCase().includes(query));
+      break;
+    default:
+      resultTodos = resultTodos.filter(todo => todo.title.toLowerCase().includes(query));
+      break;
+  }
+
+  return resultTodos;
+}
 
 export const App: React.FC = () => {
+  const [todoShown, setTodoShown] = useState<Todo>();
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosActivityFilter, setTodosActivityFilter] = useState<FilterTypes>('All');
+  const [todosQuery, setTodosQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const setActivityFilter = (filterValue: FilterTypes) => {
+    setTodosActivityFilter(filterValue);
+  };
+
+  const setQuery = (value: string) => {
+    setTodosQuery(value);
+  };
+
+  const focusOnTodo = (todo: Todo) => {
+    setTodoShown(todo);
+  };
+
+  const unFocusOnTodo = () => {
+    setTodoShown(undefined);
```

Comment:
it should be `null` not `undefined` 
`null` means that there is absence of something, `undefined` means that some variable is not yet defined.
we use null when we want to delete some value and point that there is no todo
and `undefined` means that the the code didn't reach this line